internal/coreinternal/processor/filterconfig/config.go

```diff
@@ -125,8 +125,18 @@ type MatchProperties struct {
 	// A match occurs if the span's implementation library matches at least one item in this list.
 	// This is an optional field.
 	Libraries []InstrumentationLibrary `mapstructure:"libraries"`
+
+	// SpanKinds specify the list of items to match the span kind against.
+	// A match occurs if the span's span kind matches at least one item in this list.
+	SpanKinds []string `mapstructure:"span_kinds"`
```

Comment:
With the current implementation I can write `Server`, the configuration is valid, but it will actually not match anything, since the `String()` method of the kind returns `SPAN_KIND_SERVER`. 

This can be quite confusing, so we should do validation on the strings passed to the `SpanKinds` option.